rcl/src/rcl/node.c

```diff
@@ -72,18 +79,46 @@ rcl_node_init(rcl_node_t * node, const char * name, const rcl_node_options_t * o
     allocator->allocate, "allocate not set", return RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_FOR_NULL_WITH_MSG(
     allocator->deallocate, "deallocate not set", return RCL_RET_INVALID_ARGUMENT);
+  // Make sure the node name is valid before allocating memory.
+  int validation_result = 0;
+  size_t invalid_index = 0;
+  ret = rmw_validate_node_name(name, &validation_result, &invalid_index);
+  if (ret != RMW_RET_OK) {
+    RCL_SET_ERROR_MSG(rmw_get_error_string_safe());
+    return ret;
+  }
+  if (validation_result != RMW_NODE_NAME_VALID) {
+    const char * msg = rmw_node_name_validation_result_string(validation_result);
+    if (!msg) {
+      msg = "unknown validation_result, this should not happen";
+    }
+    RCL_SET_ERROR_MSG(msg);
+    return RCL_RET_NODE_INVALID_NAME;
+  }
+  // Make sure the node namespace is valid too, but only if it is not an empty string.
+  if (strlen(namespace_) > 0) {
+    validation_result = 0;
+    invalid_index = 0;
+    ret = rmw_validate_topic_name(namespace_, &validation_result, &invalid_index);
```

Comment:
Since the `namespace_` alone is not a FQN yet (it doesn't contain the topic name at the end, right?) doesn't this need a separate validation function (`rmw_validate_namespace`)?